akka-http-core/src/main/scala/akka/http/scaladsl/UseHttp2.scala

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+/**
+ * Specify whether to support HTTP/2: never, negotiated, or always.
+ */
+sealed trait UseHttp2 extends akka.http.javadsl.UseHttp2 {
+  override def asScala = this.asInstanceOf[UseHttp2]
+}
+object Never extends UseHttp2
+object Negotiated extends UseHttp2
+object Always extends UseHttp2
```

Comment:
Agreed about moving them to a companion object.

`UseHttp2` is indeed not a very nice name. Secure/insecure and no/negotiated/always http2 are orthogonal though:

* insecure no: http h1
* insecure negotiated: h2c (http + Upgrade + h1/h2)
* insecure always: 'raw' h2 ('preexisting knowledge')
* secure no: https + h1
* secure negotiated: https + ALPN + h1/h2
* secure always: https + h2

So it might make sense to keep them as 2 separate fields instead of an 'enum' of 6?